cmd/admin-prometheus-metrics.go

```diff
@@ -30,86 +30,135 @@ import (
 	"github.com/minio/mc/pkg/probe"
 )
 
+var metricsFlags = append(metricsV3Flags,
+	cli.StringFlag{
+		Name:  "api-version",
+		Usage: "version of metrics api to use. valid values are ['v2', 'v3']. defaults to 'v2' if not specified.",
+		Value: "v2",
+	})
+
 var adminPrometheusMetricsCmd = cli.Command{
 	Name:         "metrics",
-	Usage:        "print cluster wide prometheus metrics",
+	Usage:        "print prometheus metrics",
 	OnUsageError: onUsageError,
 	Action:       mainSupportMetrics,
 	Before:       setGlobalsFromContext,
-	Flags:        globalFlags,
+	Flags:        append(globalFlags, metricsFlags...),
 	CustomHelpTemplate: `NAME:
   {{.HelpName}} - {{.Usage}}
 USAGE:
   {{.HelpName}} TARGET [METRIC-TYPE]
 
 METRIC-TYPE:
-  valid values are ['cluster', 'node', 'bucket', 'resource']. Defaults to 'cluster' if not specified.
+  valid values are
+    api-version v2 ['cluster', 'node', 'bucket', 'resource']. defaults to 'cluster' if not specified.
+    api-version v3 ["api", "system", "debug", "cluster", "ilm", "audit", "logger", "replication", "notification", "scanner"]. defaults to all if not specified.
 
 FLAGS:
   {{range .VisibleFlags}}{{.}}
   {{end}}
-EXAMPLES:
-  1. List of metrics reported cluster wide.
+EXAMPLES (v3):
+  1. API metrics
+     {{.Prompt}} {{.HelpName}} play api
+
+  2. API metrics for the bucket 'mybucket'
+     {{.Prompt}} {{.HelpName}} play api --bucket mybucket
+
+  3. System metrics
+     {{.Prompt}} {{.HelpName}} play system
+
+  4. Debug metrics
+     {{.Prompt}} {{.HelpName}} play debug
+
+  5. Cluster metrics
+     {{.Prompt}} {{.HelpName}} play cluster
+
+  6. ILM metrics
+     {{.Prompt}} {{.HelpName}} play ilm
+
+  7. Audit metrics
+     {{.Prompt}} {{.HelpName}} play audit
+
+  8. Logger metrics
+     {{.Prompt}} {{.HelpName}} play logger
+
+  9. Replication metrics
+     {{.Prompt}} {{.HelpName}} play replication
+
+  10. Replication metrics for the bucket 'mybucket'
+      {{.Prompt}} {{.HelpName}} play replication --bucket mybucket
+
+  11. Notification metrics
+      {{.Prompt}} {{.HelpName}} play notification
+
+  12. Scanner metrics
+      {{.Prompt}} {{.HelpName}} play scanner
+
+  13. List of cluster metrics (without values)
+      {{.Prompt}} {{.HelpName}} play cluster --list
```

Comment:
```suggestion
  1. API metrics
     {{.Prompt}} {{.HelpName}} --api-version v3 play api 

  2. API metrics for the bucket 'mybucket'
     {{.Prompt}} {{.HelpName}} --api-version v3 play api --bucket mybucket

  3. System metrics
     {{.Prompt}} {{.HelpName}}  --api-version v3 play system

  4. Debug metrics
     {{.Prompt}} {{.HelpName}}  --api-version v3 play debug

  5. Cluster metrics
     {{.Prompt}} {{.HelpName}}  --api-version v3 play cluster

  6. ILM metrics
     {{.Prompt}} {{.HelpName}}  --api-version v3 play ilm

  7. Audit metrics
     {{.Prompt}} {{.HelpName}}  --api-version v3 play audit

  8. Logger metrics
     {{.Prompt}} {{.HelpName}}  --api-version v3 play logger

  9. Replication metrics
     {{.Prompt}} {{.HelpName}}  --api-version v3 play replication

  10. Replication metrics for the bucket 'mybucket'
      {{.Prompt}} {{.HelpName}} --bucket mybucket --api-version v3  play replication

  11. Notification metrics
      {{.Prompt}} {{.HelpName}}  --api-version v3 play notification

  12. Scanner metrics
      {{.Prompt}} {{.HelpName}}  --api-version v3 play scanner

  13. List of cluster metrics (without values)
      {{.Prompt}} {{.HelpName}} --list  --api-version v3play cluster
```

Not sure if I have the syntax right but since these are v3 metrics, examples above need to have the key.

Reply:
thanks, fixed